packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -8,82 +30,175 @@ export enum OptimisticOracleType {
-  requester: string; // Address of the requester.
-  requestTx: string; // Transaction hash of the request.
-  proposer?: string; // Address of the proposer.
-  proposedValue?: number | boolean; // Proposed value.
-  proposeTx?: string; // Transaction hash of the proposal.
-  disputableUntil?: number; // Timestamp in ms until the request can be disputed.
-  resolvedValue?: number | boolean; // Resolved value.
-  resolveTx?: string; // Transaction hash of the resolution.
-  disputeTx?: string; // Transaction hash of the dispute.
+  readonly requestData: RequestData;
+  readonly proposalData?: ProposalData;
+  readonly resolutionData?: ResolutionData;
 }
 
 /**
  * Represents an Optimistic Oracle request.
  */
 export class OptimisticOracleRequest {
-  readonly body: string; // Human-readable request body.
-  readonly type: OptimisticOracleType; // Type of the request.
-  readonly timestamp: number; // Timestamp in seconds of the request.
-  readonly identifier: string; // Identifier of the request.
-  readonly requester: string; // Address of the requester.
-  readonly requestTx: string; // Transaction hash of the request.
-  readonly proposer?: string; // Address of the proposer.
-  readonly proposedValue?: number | boolean; // Proposed value.
-  readonly proposeTx?: string; // Transaction hash of the proposal.
-  readonly disputableUntil?: number; // Timestamp in ms until the request can be disputed.
-  readonly resolvedValue?: number | boolean; // Resolved value.
-  readonly resolveTx?: string; // Transaction hash of the resolution.
-  readonly disputeTx?: string; // Transaction hash of the dispute.
-
+  private isEventBasedFetched = false;
+  protected isEventBased = false;
   /**
    * Creates a new instance of OptimisticOracleRequest.
    * @param data The data of the request.
    */
-  constructor(data: OptimisticOracleRequestData) {
-    this.body = data.body;
-    this.type = data.type;
-    this.timestamp = data.timestamp;
-    this.identifier = data.identifier;
-    this.requester = data.requester;
-    this.requestTx = data.requestTx;
-    this.proposer = data.proposer;
-    this.proposedValue = data.proposedValue;
-    this.proposeTx = data.proposeTx;
-    this.disputableUntil = data.disputableUntil; // should be in ms
-    this.resolvedValue = data.resolvedValue;
-    this.resolveTx = data.resolveTx;
-    this.disputeTx = data.disputeTx;
+  constructor(readonly data: OptimisticOracleRequestData) {}
+
+  async fetchIsEventBased(ooV2Contract: OptimisticOracleV2Ethers): Promise<boolean> {
+    if (this.type !== OptimisticOracleType.PriceRequest) return Promise.resolve(false);
+
+    if (this.isEventBasedFetched) return Promise.resolve(this.isEventBased);
+
+    this.isEventBased = await ooV2Contract
+      .getRequest(
+        this.data.requestData.requester,
+        this.data.requestData.identifier,
+        this.data.requestData.timestamp,
+        this.data.requestData.rawBody
+      )
+      .then((r) => r[4][0]);
```

Comment:
```suggestion
      .then((r) => r.requestSettings.eventBased);
```
better to use named properties to improve readability